src/MetaDataAPI/Erc20Manager/Erc20Provider.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System.Numerics;
+using Net.Web3.EthereumWallet;
+using Net.Cache.DynamoDb.ERC20;
+using Net.Cache.DynamoDb.ERC20.Models;
+
+namespace MetaDataAPI.Erc20Manager;
+
+public class Erc20Provider : IErc20Provider
+{
+    private readonly ERC20CacheProvider provider;
+
+    public Erc20Provider()
+    {
+        provider = new ERC20CacheProvider();
+    }
+
+    public Erc20Provider(ERC20CacheProvider provider)
+    {
+        this.provider = provider;
+    }
+
+    public Erc20Token GetErc20Token(string rpcUrl, BigInteger chainId, EthereumAddress address)
```

Comment:
"MAX_SAFE_CHAIN_ID = floor( ( 2**53 - 39 ) / 2 ) = 4503599627370476"
we can use long or long as chainId